cluster-state-service/handler/api/v1/instanceapis.go

```diff
@@ -172,8 +175,24 @@ func (instanceAPIs ContainerInstanceAPIs) StreamInstances(w http.ResponseWriter,
 	ctx, cancel := context.WithCancel(context.TODO())
 	defer cancel()
 
-	instanceRespChan, err := instanceAPIs.instanceStore.StreamContainerInstances(ctx)
+	query := r.URL.Query()
+
+	entityVersion := query.Get(instanceEntityVersionKey)
+
+	if entityVersion != "" {
+		if !regex.IsEntityVersion(entityVersion) {
+			http.Error(w, invalidEntityVersionClientErrMsg, http.StatusBadRequest)
+			return
+		}
+	}
+
+	instanceRespChan, err := instanceAPIs.instanceStore.StreamContainerInstances(ctx, entityVersion)
 	if err != nil {
+		if err == rpctypes.ErrCompacted || err == rpctypes.ErrFutureRev {
```

Comment:
Much cleaner way of handling this. I've wrapped these errors in a new custom `OutOfRangeEntityVersion` error object, and detected them here.